test/runtests.jl

```diff
@@ -1,82 +1,112 @@
-                else
-                    foo, bar = rand(1:10), rand()
+        elseif arity == 2
+            @test DiffRules.hasdiffrule(M, f, 2)
+            derivs = DiffRules.diffrule(M, f, :foo, :bar)
+            @eval begin
+                let
+                    foo, bar = if $(f === :mod)
+                        rand($T) + 13, rand($T) + 5 # make sure x/y is not integer
+                    elseif $(f === :polygamma)
+                        rand(1:10), rand($T) # only supports integers as first arguments
+                    elseif $(f in (:bessely, :besselyx))
+                        # avoid singularities with finite differencing
+                        rand($T), rand($T) + $T(0.5)
+                    elseif $(f === :log)
+                        # avoid singularities with finite differencing
+                        rand($T) + $T(1.5), rand($T)
+                    elseif $(f === :^)
+                        # avoid singularities with finite differencing
+                        rand($T) + $T(0.5), rand($T)
+                    else
+                        rand($T), rand($T)
+                    end
+                    dx, dy = $(derivs[1]), $(derivs[2])
+                    if !(isnan(dx))
+                        @test dx ≈ finitediff(z -> $M.$f(z, bar), foo) rtol=1e-2 atol=1e-3
+
+                        # Check type, if applicable.
+                        if foo isa AbstractFloat
+                            @test promote_type(typeof(real(dx)), $T) === $T
+                        end
+                    end
+                    if !(isnan(dy))
+                        @test dy ≈ finitediff(z -> $M.$f(foo, z), bar) rtol=1e-2 atol=1e-3
+
+                        # Check type, if applicable.
+                        if bar isa AbstractFloat
+                            @test promote_type(typeof(real(dy)), $T) === $T
+                        end
+                    end
                 end
-                dx, dy = $(derivs[1]), $(derivs[2])
+            end
+        elseif arity == 3
+            #=
+            @test DiffRules.hasdiffrule(M, f, 3)
+            derivs = DiffRules.diffrule(M, f, :foo, :bar, :goo)
+            @eval begin
+                foo, bar, goo = randn(3)
+                dx, dy, dz = $(derivs[1]), $(derivs[2]), $(derivs[3])
                 if !(isnan(dx))
-                    @test isapprox(dx, finitediff(z -> $M.$f(z, bar), float(foo)), rtol=0.05)
+                    @test isapprox(dx, finitediff(x -> $M.$f(x, bar, goo), foo), rtol=0.05)
                 end
                 if !(isnan(dy))
-                    @test isapprox(dy, finitediff(z -> $M.$f(foo, z), bar), rtol=0.05)
+                    @test isapprox(dy, finitediff(y -> $M.$f(foo, y, goo), bar), rtol=0.05)
+                end
+                if !(isnan(dz))
+                    @test isapprox(dz, finitediff(z -> $M.$f(foo, bar, z), goo), rtol=0.05)
                 end
             end
+            =#
```

Comment:
Reverted this change as you requested @devmotion but because it's now within the for-loop, it messed up the diff.

Reply:
A bit annoying but OK :man_shrugging: